apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/JvmMemoryMetrics.java

```diff
@@ -78,5 +81,32 @@ public double get() {
                 return platformMXBean.getNonHeapMemoryUsage().getMax();
             }
         });
+
+        List<MemoryPoolMXBean> memoryPoolMXBeans = ManagementFactory.getMemoryPoolMXBeans();
+
+        for (final MemoryPoolMXBean memoryPoolMXBean : memoryPoolMXBeans) {
+            if (memoryPoolMXBean.getType() != MemoryType.HEAP) {
+                continue;
+            }
+            final Labels memoryPoolTags = Labels.Mutable.of("type", memoryPoolMXBean.getName());
+            registry.add("jvm.memory.heap.used", memoryPoolTags, new DoubleSupplier() {
+                @Override
+                public double get() {
+                    return memoryPoolMXBean.getUsage().getUsed();
+                }
+            });
+            registry.add("jvm.memory.heap.committed", memoryPoolTags, new DoubleSupplier() {
+                @Override
+                public double get() {
+                    return memoryPoolMXBean.getUsage().getCommitted();
+                }
+            });
+            registry.add("jvm.memory.heap.max", memoryPoolTags, new DoubleSupplier() {
+                @Override
+                public double get() {
+                    return memoryPoolMXBean.getUsage().getMax();
+                }
+            });
+        }
```

Comment:
As these metrics nice-to-have, please surround with try/catch, and either log error on failure or debug on success. It will help us test that we don't break anything on all JVMs we use in the integration tests (including J9)

Reply:
added try/catch